kahypar/partition/bin_packing/i_bin_packer.h

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * This file is part of KaHyPar.
+ *
+ * Copyright (C) 2020 Nikolai Maas <nikolai.maas@student.kit.edu>
+ *
+ * KaHyPar is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * KaHyPar is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with KaHyPar.  If not, see <http://www.gnu.org/licenses/>.
+ *
+******************************************************************************/
+
+#pragma once
+
+#include "kahypar/definitions.h"
+
+namespace kahypar {
+namespace bin_packing {
+// Represents the level of the applied prepacking.
+enum class BalancingLevel : uint8_t {
```

Comment:
Should this be stored in `kahypar/kahypar/partition/context_enum_classes.h`?

Reply:
Maybe also add a brief description for each of the enum values.

Reply:
I'm not sure. Unlike the enums in `kahypar/kahypar/partition/context_enum_classes.h` it is neither stored in the context (which would not work because the context of a bisection is not saved) nor is a conversion to or from a string required.

Reply:
I see, good points!